File: 算法/Golang/（1）求猴子大王的编号.go
package main

import (
	"fmt"
)

func getKingNum(n int,m int) int{

	//声明切片
	var amount []int

	//往切片中追加猴子
	for i:=0;i<n;i++  {
		amount = append(amount,i+1)
	}
	//计算第一次要删除的猴子的索引下标
	startIndex := (m-1) % len(amount)

	for true {
		if  len(amount) <= 1{
			break
		}
		//移除操作
		amount = append(amount[:startIndex],amount[startIndex+1:]...)
		//重新计算索引下标
		startIndex = (startIndex+m-1) % len(amount)
	}
	return amount[0]
}

func main()  {
	//对切片的操作
	num := getKingNum(5,2)
	fmt.Println(num)
}

File: 算法/C++/（1）求猴子大王的编号.cpp
#include <iostream>
#include <vector>

using namespace std;


int getKingNum(int n,int m) {
	
	//向量定义
	vector<int> arr(n);

	//初始化向量
	for (int i = 0; i < n; i++)
	{
		arr[i] = i + 1;
	}

	int startIndex = (m - 1) % n;


	while (true) {
		if (arr.size() <= 1)
		{
			break;
		}
		
		//删除操作
		arr.erase(arr.begin() + startIndex);
		startIndex = (startIndex + m - 1) % arr.size();
	}
	
	return arr[0];
}

int main() {

	int num = getKingNum(3, 2);
	cout << num << endl;
}